Objects/genericaliasobject.c

```diff
@@ -626,6 +639,50 @@ static PyNumberMethods ga_as_number = {
         .nb_or = _Py_union_type_or, // Add __or__ function
 };
 
+static PyObject *
+ga_iternext(gaiterobject *gi) {
+    if (gi->obj == NULL) {
+        return NULL;
+    }
+    gaobject *alias = (gaobject *) gi->obj;
+    PyObject *starred_tuple = Py_GenericAlias(alias->origin, alias->args);
+    ((gaobject * ) starred_tuple)->starred = 1;
+    Py_SETREF(gi->obj, NULL);
+    return starred_tuple;
+}
+
+static void
+ga_iter_dealloc(gaiterobject *gi) {
+    _PyObject_GC_UNTRACK(gi);
+    PyObject_GC_Del(gi);
+}
+
+static PyTypeObject Py_GenericAliasIterType = {
+    PyVarObject_HEAD_INIT(&PyType_Type, 0)
+    .tp_name = "generic_alias_iter",
+    .tp_basicsize = sizeof(gaiterobject),
+    .tp_iternext = (iternextfunc)ga_iternext,
+    .tp_dealloc = (destructor)ga_iter_dealloc,
+    .tp_flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_HAVE_GC,
+};
+
+static PyObject *
+ga_iter(PyObject *self) {
+    gaobject *alias = (gaobject *) self;
+    if ((PyTypeObject *) alias->origin != &PyTuple_Type) {
```

Comment:
I feel like this might unduly restrict future uses for this feature. Could we live without this check?

Reply:
Yeah, should be fine. Removed.